### DISCUSSION TITLE
To create a diagonal line inside a merged cell

### DISCUSSION THREAD

**Discussion by yarakos95** [lvjr/tabularray#461] [Q&A]
I'm trying to draw only diagonal lines using `\diagbox`.

When drawing a diagonal line between merged cells, it is not possible to draw a diagonal line from the upper left corner to the lower right corner as shown below. I know that `\diagbox` can be adjusted by imposing multiple options, but I couldn't figure out how to get the desired result.

I created the following as a simple example.

```latex
\documentclass[ border = 3pt ]{standalone}
\usepackage{tabularray}
\UseTblrLibrary{diagbox}
\newcommand{\onlySlash}{
  \diagbox[
    height = 2\line+\abovesep+\belowsep,
  ]{}{}%
}
\begin{document}

\begin{tblr}{
    hlines, vlines,
    columns = {
      halign = c, valign = m,
      % co = 1,
    },
    cell{2}{2} = { r = 2, c = 1 }{},
  }
  single cell & single cell \\
  merged cell & \onlySlash  \\
  merged cell & \onlySlash  \\
\end{tblr}

\end{document}
```

![diagbox-image-1](https://github.com/lvjr/tabularray/assets/75878492/608b8673-3e42-4273-807d-b55f20905fc8)

I am expecting an option specification that satisfies the following requirements.

- Column width depends on contents of cells other than `\diagbox`
- Don't specify a specific value
- Even relative column widths like `co = 1` are automatically adjusted

### Edit

I found that `height` requires two lines of `abovesep` and `belowsep` specifications. Therefore, modify it as follows.

```diff
- height = 2\line+\abovesep+\belowsep,
+ height = 2\line+2\abovesep+2\belowsep,
```

When I look at some site pages, I see a statement saying that `width` should be `\linewidth`. However, if you actually specify `width = \linewidth`, the column width will increase.

<details>
<summary>code</summary>

```latex
\documentclass[ border = 3pt ]{standalone}
\usepackage{tabularray}
\UseTblrLibrary{diagbox}
\newcommand{\onlySlash}{
  \diagbox[
    height = 2\line+2\abovesep+2\belowsep,
    %
    % Add this ↓
    width = \linewidth
  ]{}{}%
}
\begin{document}

\begin{tblr}{
    hlines, vlines,
    columns = {
      halign = c, valign = m,
      % co = 1,
    },
    cell{2}{2} = { r = 2, c = 1 }{},
  }
  single cell & single cell \\
  merged cell & \onlySlash  \\
  merged cell & \onlySlash  \\
\end{tblr}

\end{document}
```

</details>

![diagbox-image-linewidth](https://github.com/lvjr/tabularray/assets/75878492/c96f8f39-d104-4563-a03d-694c9667c597)

----------

In addition, if you check the column width of 2nd column on the PDF without `\diagbox`, you can see that it is 55pt, so I actually specified `width = 55pt`. (I didn't expect this kind of specification method, but I tried to see if it was possible.)

<details>
<summary>code</summary>

```latex
\documentclass[ border = 3pt ]{standalone}
\usepackage{tabularray}
\UseTblrLibrary{diagbox}
\newcommand{\onlySlash}{
  \diagbox[
    height = 2\line+2\abovesep+2\belowsep,
    %
    % Add this ↓
    width = 55pt
  ]{}{}%
}
\begin{document}

\begin{tblr}{
    hlines, vlines,
    columns = {
      halign = c, valign = m,
      % co = 1,
    },
    cell{2}{2} = { r = 2, c = 1 }{},
  }
  single cell & single cell \\
  merged cell & \onlySlash  \\
  merged cell & \onlySlash  \\
\end{tblr}

\end{document}
```

</details>

![diagbox-image-55pt](https://github.com/lvjr/tabularray/assets/75878492/b3788f37-671c-4e4b-b363-7d13f120135f)

----------

Also, considering the values of `leftsep`/`rightsep`, I also tried specifying the following:

<details>
<summary>code</summary>

```latex
\documentclass[ border = 3pt ]{standalone}
\usepackage{tabularray}
\UseTblrLibrary{diagbox}
\newcommand{\onlySlash}{
  \diagbox[
    height = 2\line+2\abovesep+2\belowsep,
    %
    % Add this ↓
    width = 43pt+\leftsep+\rightsep
  ]{}{}%
}
\begin{document}

\begin{tblr}{
    hlines, vlines,
    columns = {
      halign = c, valign = m,
      % co = 1,
    },
    cell{2}{2} = { r = 2, c = 1 }{},
  }
  single cell & single cell \\
  merged cell & \onlySlash  \\
  merged cell & \onlySlash  \\
\end{tblr}

\end{document}
```

</details>

![diagbox-image-43pt+leftsep+rightsep](https://github.com/lvjr/tabularray/assets/75878492/d37c7ba3-6318-4d1a-8923-e4db9ce99365)

**Comment by muzimuzhi** [comment]
See a general-purpose attempt I just posted in https://github.com/lvjr/tabularray/discussions/387#discussioncomment-8338342.

**Reply by yarakos95** [reply]
@muzimuzhi, thank you.

It is a very developmental workaround, almost a black box for me as I am unfamiliar with Expl3, but it works as expected 😄
